pkg/scheduler/eventhandlers.go

```diff
@@ -246,16 +249,41 @@ func responsibleForPod(pod *v1.Pod, profiles profile.Map) bool {
 	return profiles.HandlesSchedulerName(pod.Spec.SchedulerName)
 }
 
+const (
+	// syncedPollPeriod controls how often you look at the status of your sync funcs
+	syncedPollPeriod = 100 * time.Millisecond
+)
+
+// WaitForHandlersSync waits for EventHandlers to sync.
+// It returns true if it was successful, false if the controller should shut down
+func (sched *Scheduler) WaitForHandlersSync(ctx context.Context) error {
+	return wait.PollImmediateUntil(syncedPollPeriod,
```

Comment:
should this be `wait.PollImmediateUntilWithContext`
```suggestion
	return wait.PollImmediateUntilWithContext(ctx, syncedPollPeriod,
```

Reply:
Both function seem to be deprecated.
```
// Deprecated: This method does not return errors from context, use PollWithContextCancel.
// Note that the new method will no longer return ErrWaitTimeout and instead return errors
// defined by the context package. Will be removed in a future release.
func PollImmediateUntilWithContext(ctx context.Context, interval time.Duration, condition ConditionWithContextFunc) error {
	return poll(ctx, true, poller(interval, 0), condition)
}

```
I found #116712 during writing this pr. Can you confirm `PollImmediateUntilWithContext` is the right function?

Reply:
use PollWithContextCancel

Reply:
> use PollWithContextCancel

Yeah, the thing is there is no function named `PollWithContextCancel` now. #116712

Reply:
yeah, good catch

Reply:
So, should I use `PollImmediateUntilWithContext` in this pr or merge this first?
I think scheduler is using `wait.Poll` in many places, I can do a cleanup for this after #116712 fixed.

Reply:
you can wait... we are in code freeze anyways

Reply:
@alculquicondor @aojea I think `wait.PollUntilContextCancel` is the right function. PTAL
I can help replace the deprecated function `wait.Poll` in scheduler pkg in another pr, if it's ok.